src/main/java/com/glencoesoftware/omero/ms/image/region/ImageDataRequestHandler.java

```diff
@@ -0,0 +1,708 @@
+    }
+
+
+    /**
+     * Get Pixels information from Image IDs
+     * @param imageIds Image IDs to get Pixels information for
+     * @param iQuery Query proxy service
+     * @return Map of Image ID vs. Populated Pixels object
+     * @throws ApiUsageException
+     * @throws ServerError
+     */
+    protected Map<Long, PixelsI> retrievePixDescription(
+            IQueryPrx iQuery, List<Long> imageIds)
+                throws ApiUsageException, ServerError {
+        ScopedSpan span = Tracing.currentTracer()
+                .startScopedSpan("retrieve_pix_description");
+        try {
+            Map<String, String> ctx = new HashMap<String, String>();
+            ctx.put("omero.group", "-1");
+            span.tag("omero.image_ids", imageIds.toString());
+            // Query pulled from ome.logic.PixelsImpl and expanded to include
+            // our required Image / Plate metadata; loading both sides of the
+            // Image <--> WellSample <--> Well collection so that we can
+            // resolve our field index.
+            ParametersI params = new ParametersI();
+            params.addIds(imageIds);
+            List<IObject> pixelsList =
+                    iQuery.findAllByQuery(
+                        "select p from Pixels as p "
+                        + "join fetch p.image as i "
+                        + "left outer join fetch i.wellSamples as ws "
+                        + "left outer join fetch ws.well as w "
+                        + "left outer join fetch w.wellSamples "
+                        + "join fetch p.pixelsType "
+                        + "join fetch p.channels as c "
+                        + "join fetch c.logicalChannel as lc "
+                        + "left outer join fetch c.statsInfo "
+                        + "left outer join fetch lc.photometricInterpretation "
+                        + "left outer join fetch lc.illumination "
+                        + "left outer join fetch lc.mode "
+                        + "left outer join fetch lc.contrastMethod "
+                        + "where i.id in (:ids)", params, ctx);
+            Map<Long, PixelsI> toReturn = new HashMap<Long, PixelsI>();
+            for (IObject pixelsObj : pixelsList) {
+                PixelsI pixels = (PixelsI) pixelsObj;
+                toReturn.put(pixels.getImage().getId().getValue(), pixels);
+            }
+            return toReturn;
+        } finally {
+            span.finish();
+        }
+    }
+
+    protected EventI queryEvent(
+            IQueryPrx iQuery, Long eventId)
+                throws ApiUsageException, ServerError {
+        ScopedSpan span = Tracing.currentTracer()
+                .startScopedSpan("query_event");
+        try {
+            Map<String, String> ctx = new HashMap<String, String>();
+            ctx.put("omero.group", "-1");
+            span.tag("omero.image_id", eventId.toString());
+            // Query pulled from ome.logic.PixelsImpl and expanded to include
```

Comment:
This comment makes no sense. Method maybe not even needed if we do the wider query first as discussed above.